option optimize_for = SPEED;
option java_package = "global";
import "common.proto";

//Contains the cluster information of the requester Note: While forwarding the Global Message Please Don't change the DestinationId
message GlobalHeader {
    required int32 cluster_id = 1; //Own Cluster Id
    required int64 time = 2;
    optional int32 destination_id = 8; // ClusterId who has got Client Request
}

message GlobalMessage {
    required GlobalHeader globalHeader = 1;
    oneof payload {
        bool ping = 2; // For testing only
        string message = 3; // For testing only
        Request request = 4; // Global File Request - (READ only for now)
        Response response = 5; // Global File Response -( contains the READ's result)
        WhoIsLeader whoIsClusterLeader = 6;
        LeaderInfo clusterLeaderInfo =7;
    }
}

message WhoIsLeader{
    required string requesterIp =1;
    required int32 requesterPort =2;
}

message LeaderInfo{
    required string leaderIp =1;
    required int32 leaderPort =2;
}